tests/vanilla/store.test.tsx

```diff
@@ -592,3 +592,43 @@ it('should update derived atom even if dependances changed (#2697)', () => {
   store.set(primitiveAtom, 1)
   expect(onChangeDerived).toHaveBeenCalledTimes(1)
 })
+
+it('should preserve dependencies when reusing continuable promise', async () => {
+  const baseAtom = atom(0)
+  let promise = Promise.resolve(-1)
+  let inProgress = false
```

Comment:
I think this test is overcomplicated.
Can you simply without using `setSelf` and maybe `inProgress`?
Not depending on micro/macro tasks, but do manual control like: `resolve()` or `resolve.splice(0).forEach((fn) => fn())`

Reply:
I'm struggling to find a simpler test.
`setSelf` is used to cause a recomputation in the same microtask

Reply:
> `setSelf` is used to cause a recomputation in the same microtask.

oh, i see.